### DISCUSSION TITLE
为啥要在合并PR之前批准PR？

### DISCUSSION THREAD

**Discussion by Deep-Dark-Forest** [Meloong-Git/PCL#2014] [提问] [answered]
为啥不能直接合并？

**Comment by so1ve** [comment, marked as answer]
因为在贡献代码之后项目维护者需要审查这个PR，比如有没有潜在的bug，代码格式是否过关，ci和测试能不能通过等等等等。如果有上述问题就让贡献者修改直到改好，然后批准一下，就能合并了。当然这是大项目，小项目如果你写着玩那当然无所谓。像Vite那样的项目一般需要两个人来审查PR，必须要两个人同时批准才能合并，这样能保证代码质量

**Reply by Deep-Dark-Forest** [reply]
> 因为在贡献代码之后项目维护者需要审查这个PR，比如有没有潜在的bug，代码格式是否过关，ci和测试能不能通过等等等等。如果有上述问题就让贡献者修改直到改好，然后批准一下，就能合并了。当然这是大项目，小项目如果你写着玩那当然无所谓。像Vite那样的项目一般需要两个人来审查PR，必须要两个人同时批准才能合并，这样能保证代码质量

事实上你说的可能不完全对。我怎么觉得是一个PR龙猫觉得行就点批准，等到版本更新的时候再合并呢

**Reply by iPodtouch6557** [reply]
还有一个原因，也许是 #91 ……
虽然可能有点内涵（动词），但是这也是确保代码质量的一部分。我只是在说事，别联系到内涵那个动词（）

**Reply by so1ve** [reply]
@deep-dark-forest 事实上是因为这个仓库的PR都没有什么太大的改动，基本上龙猫看一眼觉得不行就给你关了，觉得可以就点个批准走个过场…

**Reply by Pigeon0v0** [reply]
> 还有一个原因，也许是 #91 ……
> 
> 虽然可能有点内涵（动词），但是这也是确保代码质量的一部分。我只是在说事，别联系到内涵那个动词（）
> 
> 
> 
> 

README 算个毛代码质量…

不合并就是不合并，批准是为了让你知道你这个 Pr 会合进去，不通过的直接就关了

**Reply by iPodtouch6557** [reply]
总之要确认一下的

**Comment by Chiloven945** [comment]


**Reply by so1ve** [reply]
他应该是问为什么要点一个Approve

**Reply by PlayMcBKuwu** [reply]
理解错了 他的意思是为什么 **开发者点击合并前还得按一下批准**

**Comment by allMagicNB** [comment]
直接合并，你会获得 YOLO Highlight。
